File: library/src/main/java/com/bennyhuo/android/activitystack/OnApplicationStateChangedListener.java
package com.bennyhuo.android.activitystack;

public interface OnApplicationStateChangedListener {
    /**
     * application goes background with all activities in CREATED state.
     */
    void onBackground();

    /**
     * application goes foreground with one of the activities above CREATED state.
     */
    void onForeground();

    /**
     * called when application exits abnormally.
     */
    void onTerminate(Throwable throwable);
}

File: library/src/main/java/com/bennyhuo/android/activitystack/OnApplicationStateChangedListener.kt
package com.bennyhuo.android.activitystack

interface OnApplicationStateChangedListener {
    /**
     * application goes background with all activities in CREATED state.
     */
    fun onBackground() {}

    /**
     * application goes foreground with one of the activities above CREATED state.
     */
    fun onForeground() {}

    /**
     * called when application exits abnormally.
     */
    fun onTerminate(throwable: Throwable?) {}
}